autoexec.sh

```diff
@@ -0,0 +1,26 @@
+#!/bin/bash
+
+# Setup Node Environment
+export NVM_DIR="$HOME/.nvm"
```

Comment:
I've actually been thinking about switching to [mise](https://mise.jdx.dev/) for Node, Python, and hopefully anything else we may need at some point. I've switched to it on a few machines and quite link it. Thoughts?

Reply:
It looks like it would be elegant for both node and python.

Reply:
I played with mise, and am finding it a bit frustrating.

Followed the documentation, but although I am doing:
```
mise install python@3.10
mise use -g python@3.10
mise activate bash
```

I am unable to run 'python' to verify that it is in the path.

In addition, the mise instructions state "to install npm, run: mise use -g node", however this does not result in npm being available in the PATH.

Could you share some samples?

Reply:
I haven't run into that personally, but I also haven't tried putting it into a script - did you source the full mise environment?

Reply:
Figured most of it out.

Next problem is that when I run ./autoexec.sh it will work in my shell session but not when executed as a systemctl service (will commit this soon), as I get the error:
```
[2024-11-25T03:27:19.067Z]  WARN: ENiGMA½ BBS/60764 on bbs: Exception during "moduleInitialize" (fullModulePath=/home/egonis/enigma-bbs/core/user_2fa_otp.js)
    error: The module '/home/egonis/enigma-bbs/node_modules/node-pty/build/Release/pty.node'
    was compiled against a different Node.js version using
    NODE_MODULE_VERSION 108. This version of Node.js requires
    NODE_MODULE_VERSION 109. Please try re-compiling or re-installing
    the module (for instance, using `npm rebuild` or `npm install`).
```

Feels like it's using the system node, or something...

Reply:
The service unit probably needs to point to the exact node bin `which` is under the mise install area. Might be a better scriptable way

Reply:
Binaries are here:
```
~/.local/share/mise/installs/python/${ENIGMA_PYTHON_VERSION}/bin
~/.local/share/mise/installs/node/${ENIGMA_NODE_VERSION}/bin
```